compiler/rustc_sanitizers/src/cfi/typeid/itanium_cxx_abi/transform.rs

```diff
@@ -453,6 +453,35 @@ pub(crate) fn transform_instance<'tcx>(
             instance.def = ty::InstanceKind::Virtual(call, 0);
             instance.args = abstract_args;
         }
+        let fn_traits = [
+            (LangItem::Fn, sym::call),
+            (LangItem::FnMut, sym::call_mut),
+            (LangItem::FnOnce, sym::call_once),
+        ];
+        for (lang_item, method_sym) in fn_traits {
+            if let Some(trait_id) = tcx.lang_items().get(lang_item) {
+                let items = tcx.associated_items(trait_id);
+                if let Some(call_method) =
+                    items.in_definition_order().find(|item| item.name() == method_sym)
+                {
```

Comment:
This logic does a ton of unnecessary work. 

Just compute `if let Some(trait_def_id) = tcx.trait_of_assoc(instance.def_id())`, use `tcx.as_lang_item()` to turn it into a `LangItem`, and match on `Fn`/`FnMut`/`FnOnce`.